styx-common/src/main/java/com/spotify/styx/storage/AggregateStorage.java

```diff
@@ -172,6 +172,11 @@ public void updateNextNaturalTriggerOld(WorkflowId workflowId, Instant instant)
   }
 
   @Override
+  public Map<WorkflowId, Workflow> workflows() throws IOException {
```

Comment:
We trigger workflow instance based on whether it has `nextNaturalTrigger` or not. So if it never had, we probably never created SA secret for it anyway. https://github.com/spotify/styx/blob/829da8b159c16fbd5d47982024a44b0616af5cfe/styx-scheduler-service/src/main/java/com/spotify/styx/workflow/WorkflowInitializer.java#L70-L70

I think the case is something wrong in the workflow configuration that we never managed to parse it correctly. Did I missing something?